tiny_dnn/core/kernels/tiny_quantized_fully_connected_kernel.h

```diff
@@ -40,106 +40,86 @@ inline void tiny_quantized_fully_connected_kernel(const fully_params& params,
     std::vector<uint8_t> in_quantized =
-        float_tensor_to_quantized<uint8_t>(in, min_input, max_input);
-    // filter quantization
-    float_t min_filter(W[0]);
-    float_t max_filter(W[0]);
-    for (serial_size_t c = 0; c < W.size(); c++) {
-        min_filter = std::min(min_filter, W[c]);
-        max_filter = std::max(max_filter, W[c]);
-    }
-    if (min_filter == max_filter) {
-      max_filter = W[0] + 1e-3f;
-      min_filter = W[0] - 1e-3f;
-    }
-    std::vector<uint8_t> W_quantized =
-        float_tensor_to_quantized<uint8_t>(W, min_filter, max_filter);
-    // output range
-    float_t min_output_value;
-    float_t max_output_value;
-    quantization_range_for_multiplication<uint8_t, uint8_t, int32_t>(
-        min_input, max_input, min_filter, max_filter, &min_output_value,
-        &max_output_value);
-    // bias quantization
-    float_t min_bias(0);
-    float_t max_bias(0);
-    std::vector<uint8_t> bias_quantized;
-    if (params.has_bias_) {
-        for (serial_size_t inc = 0; inc < b.size(); inc++) {
-            min_bias = std::min(min_bias, b[inc]);
-            max_bias = std::max(max_bias, b[inc]);
-        }
-        if (min_bias == max_bias) {
-          max_bias = b[0] + 1e-3f;
-          min_bias = b[0] - 1e-3f;
-        }
-        bias_quantized =
-            float_tensor_to_quantized<uint8_t>(b, min_bias, max_bias);
-    }
-    min_output_value += min_bias;
-    max_output_value += max_bias;
-
-    std::vector<int32_t> a_quantized(a.size(), static_cast<int32_t>(0));
-
-    // calculating offset
+        float_tensor_to_quantized<uint8_t>(in, range_in[0], range_in[1]);
     const int32_t offset_input =
-        float_to_quantized_unclamped<uint8_t>(0.0f, min_input, max_input);
+        float_to_quantized_unclamped<uint8_t>(0.0f, range_in[0], range_in[1]);
+
+    // Filter quantization
+    vec_t range_w = tensor_range<float_t>(W);
+    std::vector<uint8_t> w_quantized =
+        float_tensor_to_quantized<uint8_t>(W, range_w[0], range_w[1]);
     const int32_t offset_filter =
-        float_to_quantized_unclamped<uint8_t>(0.0f, min_filter, max_filter);
+        float_to_quantized_unclamped<uint8_t>(0.0f, range_w[0], range_w[1]);
+
+    // Output range
+    vec_t range_output(2, 0);
+    quantization_range_for_multiplication<uint8_t, uint8_t, int32_t>(
+        range_in[0], range_in[1], range_w[0], range_w[1],
+        &(range_output[0]), &(range_output[1]));
+    std::vector<int32_t> a_quantized(a.size(), static_cast<int32_t>(0));
     const int32_t zero_in_total_space =
-        float_to_quantized<int32_t>(0.0f, min_output_value, max_output_value);
+        float_to_quantized<int32_t>(0.0f, range_output[0], range_output[1]);
 
-    const int32_t offset_output = 0;
-    const int32_t mult_output = 1;
-    const int32_t shift_output = 0;
+    const int32_t offset_output = 0, mult_output = 1, shift_output = 0;
 
+    // Calculating inner product
     bool use_gemm = false;
```

Comment:
@wangyida I see. Thank you for the info. So, it looks like `gemmlowp` is currently off.